clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/InternalLocalBootstrappingVeniceChangelogConsumer.java

```diff
@@ -518,7 +520,13 @@ public CompletableFuture<Void> start(Set<Integer> partitions) {
   @Override
   public CompletableFuture<Void> start() {
     Set<Integer> allPartitions = new HashSet<>();
-    for (int partition = 0; partition < partitionCount; partition++) {
+    try {
+      storeRepository.subscribe(storeName);
+    } catch (InterruptedException e) {
+      throw new VeniceException("Failed to start bootstrapping changelog consumer with error:", e);
+    }
+    Store store = storeRepository.getStore(storeName);
+    for (int partition = 0; partition < store.getPartitionCount(); partition++) {
```

Comment:
`ReadOnlyViewStore` did not override the `getPartitionCount()` method so `store.getPartitionCount()` will return the original store's partition count even for views. Is this the desired behavior?

Reply:
That is not the desired behavior.  I guess I was hoping the abstraction would return the view partition count.